src/ray/raylet/lineage_cache.cc

```diff
@@ -286,7 +290,7 @@ Lineage LineageCache::GetUncommittedLineage(const TaskID &task_id,
         // The stopping condition for recursion is that the entry has
         // been committed to the GCS or has already been forwarded.
         // The lineage always includes the requested task id.
-        return entry.WasExplicitlyForwarded(node_id) && !(entry.GetEntryId() == task_id);
```

Comment:
Doesn't removing this reintroduce the bug where the lineage didn't include the requested task id if it was forwarded before, or is that no longer necessary to have?

Reply:
Oops, nice catch! I was debugging something and left this in the commit by accident...

Reply:
Hmm `LineageCacheTest#TestMarkTaskAsForwarded` has an `assert` that should catch the error if that line is missing, did it not get flagged? I can fix the test if not.

Reply:
Ah, good point! I think we also want to check `forwarded_task_id` in the test you have [here](https://github.com/ray-project/ray/blob/master/src/ray/raylet/lineage_cache_test.cc#L219). I can go ahead and add that to this PR.